File: happy birthday/birthday.go
// A try of OOP style.
package main

import (
  "fmt"
  "time"
  "strconv"
)

// Take name, birth, age and birthday
type personInfo struct {
  name string
  birth time.Time
  age int
  birthday bool
}

// Congrats birthday of other person
func (this *personInfo) congratsBirthday(person personInfo) string {
  if person.birthday {
    return this.name + ": Happy birthday for your " + strconv.Itoa(person.age) + " years! " + person.name + "!"
  } else {
    return "Today is not " + person.name + "'s birthday."
  }
}

// A function that calculate actual age
func calculateAge(birth time.Time) int {
  today := time.Now()
  var compareDate int

  if today.Month() < birth.Month() || (today.Month() == birth.Month() && today.Day() < birth.Day()) {
    compareDate = 1
  } else {
    compareDate = 0
  }

  return today.Year() - birth.Year() - compareDate
}

// Check if today is birthday
func checkBirtday(birth time.Time) bool {
  today := time.Now()
  if today.Day() == birth.Day() && today.Month() == birth.Month() {
    return true
  } else {
    return false
  }
}

func main() {
  pepeBirth := time.Date(2001, 11, 9, 0, 0, 0, 0, time.UTC)
  manoloBirth := time.Date(2000, 7, 23, 0, 0, 0, 0, time.UTC)

  pepe := personInfo{"Pepe", pepeBirth, calculateAge(pepeBirth), checkBirtday(pepeBirth)}
  manolo := personInfo{"Manolo", manoloBirth, calculateAge(manoloBirth), checkBirtday(manoloBirth)}

  fmt.Println(manolo.congratsBirthday(pepe))
}

File: happy birthday/birthday.py
# OOP birthday congrats
from datetime import date

# A class that takes name and birth date (optional), calculate age and check birthday.
class Person:
  def __init__(self, name, birth = False):
    self.name = name

    if birth:
        today = date.today()
        if (today.month < birth.month or (today.month == birth.month and today.day < birth.day)):
          compare_date = 1
        else: 
          compare_date = 0
        self.birth = birth
        self.birthday = today.day == birth.day and today.month == birth.month
        self.age = today.year - birth.year - compare_date

  # Congrats birthday of a other person.
  def congrats_birthday(self, person):
    if person.birthday:
      return f"{self.name}: Happy birthday for your {person.age} years! {person.name}!"
    else:
      return f"Today is not {person.name}'s birthday."

Manolo = Person("Manolo")
Pepe = Person("Pepe", date(2001, 11, 10))

print(Manolo.congrats_birthday(Pepe), Pepe.age)